auth/oidc.go

```diff
@@ -245,7 +246,7 @@ func (opm OIDCProviderMap) Stop() {
 func (op *OIDCProvider) GetClient(buildCallbackURLFunc OIDCCallbackURLFunc) *OIDCClient {
 	// Initialize the client on first request. If the callback URL isn't defined for the provider,
 	// uses buildCallbackURLFunc to construct (based on current request)
-	op.clientOnce.Do(func() {
+	if op.clientOnce.CompareAndSwap(false, true) { // If false, make true
```

Comment:
Could instead check if false, and then change to true when the OIDC client is set successfully so if this is called concurrently, before the OIDC Client is initialized, then nil is not returned.

The current CompareAndSwap way more emulates the old behaviour with the difference that it can be retried if the client failed to get initialized. 

Would like to hear your thoughts

Reply:
I don't think this follows the old behaviour.

`sync.Once` blocks concurrent requests until the function has finished, allowing all of the requests to use a client once the first has initialised it.

I think this blocking is the behaviour we _want_ to do, as even checking for false is not sufficinet to allow simultaenous requests to initialise and use a client for all of the requests.

The comment in CBG-1378 describes the approach we want to take, using a combination of atomic and mutex, in the same way that sync.Once is doing under the hood.

Reply:
A key feature of sync.Once.Do() is that concurrent invocations will block until the initial call completes.  We'll need to maintain the same functionality with the replacement.  The approach you've got here doesn't address that scenario, as far as I can tell.